### PR TITLE
[extractor/iwara] Overhaul extractors

### LINKED ISSUE #3851: [BUG] Iwara.tv: unable to extract upload_date
### Checklist

- [X] I'm reporting a broken site
- [X] I've verified that I'm running yt-dlp version **2022.05.18** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/ytdl-org/youtube-dl#video-url-contains-an-ampersand-and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

_No response_

### Description

Iwara.tv: unable to extract upload_date

### Verbose log

```shell
F:\downloader\iwara.tv\dl>//ytdl-patched https://www.iwara.tv/videos/axgjgi1pkupwgz4k?playlist=107712 -f best
The filename, directory name, or volume label syntax is incorrect.

F:\downloader\iwara.tv\dl>yt-dlp -a 000MMD_links.txt --conf config_iwara -v
[debug] Command-line config: ['-a', '000MMD_links.txt', '--conf', 'config_iwara', '-v']
[debug] | Config "config_iwara": ['--continue', '--http-chunk-size', '1M', '--buffer-size', '16K', '--sub-lang', 'english', '--embed-metadata', '--embed-thumbnail', '--abort-on-unavailable-fragment', '-o', '%(age_limit)s\\%(uploader)s\\%(title)s [%(id)s].%(ext)s', '--download-archive', 'downloaded.txt', '-f', 'Source/540p/360p', '--ffmpeg-location', 'D:\\Program Files\\ffmpeg\\bin']
[debug] User config "C:\Users\X\AppData\Roaming\yt-dlp\config": ['--continue', '--format-sort', 'vbr,abr', '--http-chunk-size', '1M', '--buffer-size', '16K', '--sub-lang', 'english', '--embed-metadata', '--abort-on-unavailable-fragment', '-o', '%(title)s [%(id)s].%(ext)s', '--ffmpeg-location', 'D:\\Program Files\\ffmpeg\\bin']
[debug] Batch file urls: ['https://ecchi.iwara.tv/videos/03yedh9lbizq1kq8?language=en', 'https://ecchi.iwara.tv/videos/18kqetlxjt3w8wd3?language=en', 'https://ecchi.iwara.tv/videos/25xbni92aixod0wo?language=en', 'https://ecchi.iwara.tv/videos/4jqbmudrwcwmoqwl?language=en', 'https://ecchi.iwara.tv/videos/6a8als43auo2aezk?language=en', 'https://ecchi.iwara.tv/videos/6zq9zc43auol8kbv?language=en', 'https://ecchi.iwara.tv/videos/7v5rxslm6hgrq3ao?language=en', 'https://ecchi.iwara.tv/videos/88jg3t32ytlbzyvk?language=en', 'https://ecchi.iwara.tv/videos/9zopqhkyxhzv2omq?language=en', 'https://ecchi.iwara.tv/videos/aljzzhvylt72jrgy?language=en', 'https://ecchi.iwara.tv/videos/bnawwf4byu38xepq?language=en', 'https://ecchi.iwara.tv/videos/bzq64f4byu3gvbkq?language=en', 'https://ecchi.iwara.tv/videos/deq7wfo2btkdlo2b?language=en', 'https://ecchi.iwara.tv/videos/dmvr7io2btklywam?language=en', 'https://ecchi.iwara.tv/videos/ekvzasjv9umm2gkk?language=en', 'https://ecchi.iwara.tv/videos/gklk6soqzt7zw2pa?language=en', 'https://ecchi.iwara.tv/videos/gl7gdfxovhovz05p?language=en', 'https://ecchi.iwara.tv/videos/gmjyjuoqzt7a9zkk?language=en', 'https://ecchi.iwara.tv/videos/j006wtz5ytej9jjn?language=en', 'https://ecchi.iwara.tv/videos/j090xiz5ytejvmq1?language=en', 'https://ecchi.iwara.tv/videos/kapz8ukgvhqwqme0?language=en', 'https://ecchi.iwara.tv/videos/kxymrhkgvhq5ol2a?language=en', 'https://ecchi.iwara.tv/videos/l0jb6ckwlh8oamp1?language=en', 'https://ecchi.iwara.tv/videos/ldyazhl85hwr3pvr?language=en', 'https://ecchi.iwara.tv/videos/lwnzwhkwlh8grypd?language=en', 'https://ecchi.iwara.tv/videos/lyo9rukwlh8kwkgb?language=en', 'https://ecchi.iwara.tv/videos/make3crjmu8y4klg?language=en', 'https://ecchi.iwara.tv/videos/mjr0xc3bntozmmgb?language=en', 'https://ecchi.iwara.tv/videos/nmpaatgyjtpbzkml?language=en', 'https://ecchi.iwara.tv/videos/ovalos6z0c18aqjy?language=en', 'https://ecchi.iwara.tv/videos/pxaxbsd6vcok2emp?language=en', 'https://ecchi.iwara.tv/videos/q8qy1sy2xtlpjvzd?language=en', 'https://ecchi.iwara.tv/videos/qm7jqhy2xtlvmodo?language=en', 'https://ecchi.iwara.tv/videos/qykebcgzasrlyewa?language=en', 'https://ecchi.iwara.tv/videos/vo529hrdgue9oeaq?language=en', 'https://ecchi.iwara.tv/videos/voyk4srdgue9k0r6?language=en', 'https://ecchi.iwara.tv/videos/wklyehgnlsymomol?language=en', 'https://ecchi.iwara.tv/videos/wpgbkh1xquqm0wjp?language=en', 'https://ecchi.iwara.tv/videos/wzj0kignlsyla75x?language=en', 'https://ecchi.iwara.tv/videos/xk2qjiyd5uwr1n6b?language=en', 'https://ecchi.iwara.tv/videos/2w9y0h92aiqajv7n?language=en', 'https://ecchi.iwara.tv/videos/45bz4hdrwcyjxgaz?language=en', 'https://ecchi.iwara.tv/videos/5vob5sboghwx7j4m?language=en', 'https://ecchi.iwara.tv/videos/6r7g5i43auonmo3q?language=en', 'https://ecchi.iwara.tv/videos/7oywzslm6hmmjo48?language=en', 'https://ecchi.iwara.tv/videos/ak2bouvylt7k4edb?language=en', 'https://ecchi.iwara.tv/videos/bkmjgh4byu34qrra?language=en', 'https://ecchi.iwara.tv/videos/g8pqwhxovhowvxbg?language=en', 'https://ecchi.iwara.tv/videos/gpaqpfoqzt7z4be8?language=en', 'https://ecchi.iwara.tv/videos/gprj1toqzt7grj5j?language=en', 'https://ecchi.iwara.tv/videos/j9vq2sz5yte7qq4z?language=en', 'https://ecchi.iwara.tv/videos/k9wkwckgvhqa9zjr?language=en', 'https://ecchi.iwara.tv/videos/kky6qszmbtzevgz6?language=en', 'https://ecchi.iwara.tv/videos/lkkwmhl85hw6vdoy?language=en', 'https://ecchi.iwara.tv/videos/lnllkcl85hw7zaaz?language=en', 'https://ecchi.iwara.tv/videos/lqxazukwlh8rv6md?language=en', 'https://ecchi.iwara.tv/videos/nab1vigyjtp8wmpb?language=en', 'https://ecchi.iwara.tv/videos/o13opi6z0c1wwwjy?language=en', 'https://ecchi.iwara.tv/videos/o6v4jfrj9t6rqwze?language=en', 'https://ecchi.iwara.tv/videos/0npgys9lbizoeeeq?language=en', 'https://ecchi.iwara.tv/videos/57q8yiboghwkevg8?language=en', 'https://ecchi.iwara.tv/videos/5adrycboghwrx8m9?language=en', 'https://ecchi.iwara.tv/videos/86wrru32ytl3vk93?language=en', 'https://ecchi.iwara.tv/videos/87pkbs32ytlv3jrb?language=en', 'https://ecchi.iwara.tv/videos/a7jo6umbvt8ryvj0?language=en', 'https://ecchi.iwara.tv/videos/ag5vjivylt75wzer?language=en', 'https://ecchi.iwara.tv/videos/b7nwjh4byu39aw6g?language=en', 'https://ecchi.iwara.tv/videos/blm0wc4byu3ybwbr?language=en', 'https://ecchi.iwara.tv/videos/blp3eiv5dtlnv1ar?language=en', 'https://ecchi.iwara.tv/videos/delknho2btk362zx?language=en', 'https://ecchi.iwara.tv/videos/delmnco2btk3oq73?language=en', 'https://ecchi.iwara.tv/videos/dje7buo2btkknlow?language=en', 'https://ecchi.iwara.tv/videos/e41yzsjv9umggyxb?language=en', 'https://ecchi.iwara.tv/videos/ejakwivywtk1am3y?language=en', 'https://ecchi.iwara.tv/videos/exdjdijv9umrmbrw?language=en', 'https://ecchi.iwara.tv/videos/gmwgltxovhoblogy?language=en', 'https://ecchi.iwara.tv/videos/gwewqfxovhogl7bd?language=en', 'https://ecchi.iwara.tv/videos/jlymasdeycwq5p4v?language=en', 'https://ecchi.iwara.tv/videos/jlzyasdeycw7ygqy?language=en', 'https://ecchi.iwara.tv/videos/k40znuzmbtzbon0b?language=en', 'https://ecchi.iwara.tv/videos/kdkgeikgvhqlbjgj?language=en', 'https://ecchi.iwara.tv/videos/kg76kfkgvhqzpkro?language=en', 'https://ecchi.iwara.tv/videos/komebuzmbtzmxlaq?language=en', 'https://ecchi.iwara.tv/videos/odedzt6z0c1j89gm?language=en', 'https://ecchi.iwara.tv/videos/pg5eosd6vcow4j8z?language=en', 'https://ecchi.iwara.tv/videos/q9jlrcy2xtl6llve?language=en', 'https://ecchi.iwara.tv/videos/qjbq7cgzasrjagll?language=en', 'https://ecchi.iwara.tv/videos/qnqm2cgzasr3njlr?language=en', 'https://ecchi.iwara.tv/videos/qnyj1igzasrbbojy?language=en', 'https://ecchi.iwara.tv/videos/r524liklrso8lm7z?language=en', 'https://ecchi.iwara.tv/videos/rga4ltyr9uzgezkr?language=en', 'https://ecchi.iwara.tv/videos/rrx5zcyr9uz4bdzo?language=en', 'https://ecchi.iwara.tv/videos/vb0ems2vkuegaqgz?language=en', 'https://ecchi.iwara.tv/videos/vpzozcrdgue2oorg?language=en', 'https://ecchi.iwara.tv/videos/wbd0bugnlsy0p6ow?language=en', 'https://ecchi.iwara.tv/videos/wglxvfgnlsynkr1v?language=en', 'https://ecchi.iwara.tv/videos/xob2xfyd5uwqwlvn?language=en', 'https://ecchi.iwara.tv/videos/z9xqkfa4aigwjvpe?language=en', 'https://ecchi.iwara.tv/videos/zanowta4aig2r4p3?language=en', 'https://ecchi.iwara.tv/videos/oa39vcrj9t6bjqwa?language=en', 'https://ecchi.iwara.tv/videos/ozdkwt6z0c1kqnov?language=en', 'https://ecchi.iwara.tv/videos/pwqy5hd6vcowg6q8?language=en', 'https://ecchi.iwara.tv/videos/q4z24sgzasr5bobm?language=en', 'https://ecchi.iwara.tv/videos/q6b4yhgzasrmro00?language=en', 'https://ecchi.iwara.tv/videos/qd3zmiy2xtl3mv8y?language=en', 'https://ecchi.iwara.tv/videos/qvmq1hy2xtllezjw?language=en', 'https://ecchi.iwara.tv/videos/rbagwuklrsokopwj?language=en', 'https://ecchi.iwara.tv/videos/rkm4nuyr9uzeevr9?language=en', 'https://ecchi.iwara.tv/videos/rmbqviklrso6dy1p?language=en', 'https://ecchi.iwara.tv/videos/v7yaltrdguev6l3z?language=en', 'https://ecchi.iwara.tv/videos/valw3i2vkuejmmj1?language=en', 'https://ecchi.iwara.tv/videos/vazzpf2vkuejylan?language=en', 'https://ecchi.iwara.tv/videos/vkxazs2vkuer5da1?language=en', 'https://ecchi.iwara.tv/videos/wnbwyi1xquqxym32?language=en', 'https://ecchi.iwara.tv/videos/wx5eatgnlsybdlkl?language=en', 'https://ecchi.iwara.tv/videos/wxoqdignlsyel6v4?language=en', 'https://ecchi.iwara.tv/videos/wxrelignlsya8omd?language=en', 'https://ecchi.iwara.tv/videos/ydrmmh9yjio8qaer?language=en', 'https://ecchi.iwara.tv/videos/z5njbf4anukwdkla?language=en', 'https://ecchi.iwara.tv/videos/zj9mvca4aigk2ekq?language=en', 'https://ecchi.iwara.tv/videos/0e4qzh9lbizye9q9?language=en', 'https://ecchi.iwara.tv/videos/0wlxrf9lbizd5kb2?language=en', 'https://ecchi.iwara.tv/videos/0z0eyu9lbizob4gl?language=en', 'https://ecchi.iwara.tv/videos/2z5pys92aiq43b7x?language=en', 'https://ecchi.iwara.tv/videos/5qg88hboghw0krwe?language=en', 'https://ecchi.iwara.tv/videos/84ygqs32ytloraqo?language=en', 'https://ecchi.iwara.tv/videos/8evjgf32ytlyazyo?language=en', 'https://ecchi.iwara.tv/videos/9e2kpikyxhzqorzy?language=en', 'https://ecchi.iwara.tv/videos/b5jjosv5dtlyjgwk?language=en', 'https://ecchi.iwara.tv/videos/dawbvfo2btkgxyqm?language=en', 'https://ecchi.iwara.tv/videos/djj8wuo2btk9oln9?language=en', 'https://ecchi.iwara.tv/videos/erzbwsvywtk9gqa3?language=en', 'https://ecchi.iwara.tv/videos/ex3rmhvywtkw88p9?language=en', 'https://ecchi.iwara.tv/videos/glkevioqzt7rx2yv?language=en', 'https://ecchi.iwara.tv/videos/j8exesdeycweay1y?language=en', 'https://ecchi.iwara.tv/videos/jjaaytz5yte4gwgk?language=en', 'https://ecchi.iwara.tv/videos/jjr3xuz5yte47m9z?language=en', 'https://ecchi.iwara.tv/videos/l8xvvtl85hwv82ve?language=en', 'https://ecchi.iwara.tv/videos/mbj9zcrjmu8gjwkr?language=en', 'https://ecchi.iwara.tv/videos/mervefrjmu84d0lk?language=en', 'https://ecchi.iwara.tv/videos/nmmxnsgyjtp7yql7?language=en', 'https://ecchi.iwara.tv/videos/ogywlc6z0c19aa7v?language=en', 'https://ecchi.iwara.tv/videos/orzwat6z0c1xv8jz?language=en', 'https://ecchi.iwara.tv/videos/q9lywsgzasreaqbv?language=en', 'https://ecchi.iwara.tv/videos/qemwvfy2xtl8z6lo?language=en', 'https://ecchi.iwara.tv/videos/v4rr6i2vkuewo44n?language=en', 'https://ecchi.iwara.tv/videos/vgmbwt2vkueoyrpo?language=en', 'https://ecchi.iwara.tv/videos/vloybfrdgueedelb?language=en', 'https://ecchi.iwara.tv/videos/wbgbgfgnlsy6epyl?language=en', 'https://ecchi.iwara.tv/videos/wn92jf1xquq132ge?language=en', 'https://ecchi.iwara.tv/videos/zn7dxu4anukygxrv?language=en']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.05.18 [b14d52355]
[debug] Lazy loading extractors is disabled
[debug] Python version 3.10.4 (CPython 64bit) - Windows-10-10.0.19043-SP0
[debug] Checking exe version: "D:\Program Files\ffmpeg\bin\ffmpeg" -bsfs
[debug] Checking exe version: "D:\Program Files\ffmpeg\bin\ffprobe" -bsfs
[debug] exe versions: ffmpeg 2022-03-28-git-5ee198f9aa-essentials_build-www.gyan.dev (setts), ffprobe 2022-03-28-git-5ee198f9aa-essentials_build-www.gyan.dev
[debug] Optional libraries: Cryptodome-3.14.1, brotli-1.0.9, certifi-2021.10.08, mutagen-1.45.1, sqlite3-2.6.0, websockets-10.2
[debug] Proxy map: {}
[debug] Loading archive file 'downloaded.txt'
[debug] [Iwara] Extracting URL: https://ecchi.iwara.tv/videos/03yedh9lbizq1kq8?language=en
[Iwara] 03yedh9lbizq1kq8: Downloading webpage
[Iwara] 03yedh9lbizq1kq8: Downloading JSON metadata
WARNING: [Iwara] unable to extract upload_date; please report this issue on  https://github.com/yt-dlp/yt-dlp/issues?q= , filling out the appropriate issue template. Confirm you are on the latest version using  yt-dlp -U
[debug] Sort order given by user: vbr, abr
[debug] Formats sorted by: hasvid, ie_pref, vbr, abr, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), acodec, filesize, fs_approx, tbr, asr, proto, vext, aext, hasaud, source, id
[download] SOLAR - Spit it out has already been recorded in the archive
```

### LINKED ISSUE #6549: iwara.tv update broke downloading
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that a **supported** site is broken
- [X] I've verified that I'm running yt-dlp version **2023.03.04** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

_No response_

### Provide a description that is worded well enough to be understood

iwara.tv updated their ui which changed how videos are served, making video and channel downloading not work. Probably everything for the site needs to be updated.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
~$ yt-dlp -vU https://ecchi.iwara.tv/video/1KR8Z8OI4sG1Qb
[debug] Command-line config: ['-vU', 'https://ecchi.iwara.tv/video/1KR8Z8OI4sG1Qb']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.03.04 [392389b7d]
[debug] Lazy loading extractors is disabled
[debug] Python 3.11.2 (CPython x86_64 64bit) - Linux-6.1.18-200.fc37.x86_64-x86_64-with-glibc2.36 (OpenSSL 3.0.8 7 Feb 2023, glibc 2.36)
[debug] exe versions: ffmpeg 5.1.2 (setts), ffprobe 5.1.2
[debug] Optional libraries: Cryptodome-3.17.0, brotli-1.0.9, certifi-2021.10.08, mutagen-1.45.1, sqlite3-2.6.0, websockets-10.3
[debug] Proxy map: {}
[debug] Loaded 1786 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.03.04, Current version: stable@2023.03.04
yt-dlp is up to date (stable@2023.03.04)
[generic] Extracting URL: https://ecchi.iwara.tv/video/1KR8Z8OI4sG1Qb
[generic] 1KR8Z8OI4sG1Qb: Downloading webpage
WARNING: [generic] Falling back on generic information extractor
[generic] 1KR8Z8OI4sG1Qb: Extracting information
[debug] Looking for embeds
ERROR: Unsupported URL: https://ecchi.iwara.tv/video/1KR8Z8OI4sG1Qb
Traceback (most recent call last):
  File "/usr/lib/python3.11/site-packages/yt_dlp/YoutubeDL.py", line 1518, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/yt_dlp/YoutubeDL.py", line 1594, in __extract_info
    ie_result = ie.extract(url)
                ^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/yt_dlp/extractor/common.py", line 694, in extract
    ie_result = self._real_extract(url)
                ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/yt_dlp/extractor/generic.py", line 2510, in _real_extract
    raise UnsupportedError(url)
yt_dlp.utils.UnsupportedError: Unsupported URL: https://ecchi.iwara.tv/video/1KR8Z8OI4sG1Qb

~$ yt-dlp -vU https://ecchi.iwara.tv/profile/mmdnecrox
[debug] Command-line config: ['-vU', 'https://ecchi.iwara.tv/profile/mmdnecrox']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.03.04 [392389b7d]
[debug] Lazy loading extractors is disabled
[debug] Python 3.11.2 (CPython x86_64 64bit) - Linux-6.1.18-200.fc37.x86_64-x86_64-with-glibc2.36 (OpenSSL 3.0.8 7 Feb 2023, glibc 2.36)
[debug] exe versions: ffmpeg 5.1.2 (setts), ffprobe 5.1.2
[debug] Optional libraries: Cryptodome-3.17.0, brotli-1.0.9, certifi-2021.10.08, mutagen-1.45.1, sqlite3-2.6.0, websockets-10.3
[debug] Proxy map: {}
[debug] Loaded 1786 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.03.04, Current version: stable@2023.03.04
yt-dlp is up to date (stable@2023.03.04)
[generic] Extracting URL: https://ecchi.iwara.tv/profile/mmdnecrox
[generic] mmdnecrox: Downloading webpage
WARNING: [generic] Falling back on generic information extractor
[generic] mmdnecrox: Extracting information
[debug] Looking for embeds
ERROR: Unsupported URL: https://ecchi.iwara.tv/profile/mmdnecrox
Traceback (most recent call last):
  File "/usr/lib/python3.11/site-packages/yt_dlp/YoutubeDL.py", line 1518, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/yt_dlp/YoutubeDL.py", line 1594, in __extract_info
    ie_result = ie.extract(url)
                ^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/yt_dlp/extractor/common.py", line 694, in extract
    ie_result = self._real_extract(url)
                ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/yt_dlp/extractor/generic.py", line 2510, in _real_extract
    raise UnsupportedError(url)
yt_dlp.utils.UnsupportedError: Unsupported URL: https://ecchi.iwara.tv/profile/mmdnecrox
```

### PR DESCRIPTION
**Pull request by Lesmiscore**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

ADD DESCRIPTION HERE

Fixes #6549
Fixes #3851 (no longer applies after website renewal)

Notes
- Website is now unstable. Extractions may fail
- Due to migration, only recent videos (uploaded after Mar 10?) do play. Older videos do not play on website and yt-dlp gives `No video formats found!` error

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

File: yt_dlp/extractor/iwara.py [hunk 1]
```diff
@@ -1,239 +1,190 @@
...
+            'timestamp': 1678732213,
+            'modified_timestamp': 1679110271,
         },
-        'add_ie': ['Youtube'],
     }]
 
+    def _extract_formats(self, video_id, fileurl):
+        up = urllib.parse.urlparse(fileurl)
+        q = urllib.parse.parse_qs(up.query)
+        paths = up.path.split('/')
```
Comment (pukkandan) [inline yt_dlp/extractor/iwara.py:61]: What if URL ends in `/`?
```suggestion
        paths = up.path.rstrip('/').split('/')
```

File: yt_dlp/extractor/iwara.py [hunk 2]
```diff
@@ -1,239 +1,190 @@
...
+        # https://github.com/yt-dlp/yt-dlp/issues/6549#issuecomment-1473771047
+        x_version = hashlib.sha1('_'.join((paths[-1], q['expires'][0], '5nFp9kmbNnHdAFhaqMvt')).encode()).hexdigest()
+
+        files = self._download_json(fileurl, video_id, headers={'X-Version': x_version})
+        for fmt in files:
+            yield traverse_obj(fmt, {
+                'format_id': 'name',
+                'url': ('src', ('view', 'download'), {lambda x: self._proto_relative_url(x, 'https:')}),
```
Comment (pukkandan) [inline yt_dlp/extractor/iwara.py:69]:
```suggestion
                'url': ('src', ('view', 'download'), {self._proto_relative_url}),
```
should be enough, no?

File: yt_dlp/extractor/iwara.py [hunk 3]
```diff
@@ -1,239 +1,190 @@
...
+                'ext': ('type', {mimetype2ext}),
+                'preference': ('name', {lambda x: int(x) if x.isdigit() else 1e4}),
```
Comment (pukkandan) [inline yt_dlp/extractor/iwara.py:71]: Pls explain why this is `preference` and not just `quality`. Remember, preference takes priority over any `-S` user gives
```suggestion
                'preference': ('name', {lambda x: int_or_none(x) or 1e4}),
```

File: yt_dlp/extractor/iwara.py [hunk 4]
```diff
@@ -1,239 +1,190 @@
...
+                'height': ('name', {lambda x: int(x)}),
```
Comment (pukkandan) [inline yt_dlp/extractor/iwara.py:72]:
```suggestion
                'height': ('name', {int_or_none}),
```

File: yt_dlp/extractor/iwara.py [hunk 5]
```diff
@@ -1,239 +1,190 @@
...
+                'like_count': 'numLikes',
+                'view_count': 'numViews',
+                'comment_count': 'numComments',
+                'timestamp': ('createdAt', {unified_timestamp}),
+                'modified_timestamp': ('updatedAt', {unified_timestamp}),
+                'thumbnail': (
+                    'file', 'id', {str},
+                    {lambda x: f'https://files.iwara.tv/image/thumbnail/{x}/thumbnail-00.jpg'}),
+            }),
+            'formats': list(self._extract_formats(video_id, video_data.get('fileUrl'))),
```
Comment (pukkandan) [inline yt_dlp/extractor/iwara.py:97]:
```suggestion
                'thumbnail': ('file', 'id', {str}, {
                    lambda x: f'https://files.iwara.tv/image/thumbnail/{x}/thumbnail-00.jpg'}),
            }),
            'formats': list(self._extract_formats(video_id, video_data.get('fileUrl'))),
```

File: yt_dlp/extractor/iwara.py [hunk 6]
```diff
@@ -1,239 +1,190 @@
...
+            note=f'Downloading page {page}',
+            query={
+                'page': page,
+                'sort': 'date',
+                'user': user_id,
+                'limit': self._PER_PAGE,
+            })
+        yield from (
+            self.url_result(f'https://iwara.tv/video/{x}')
+            for x in traverse_obj(videos, ('results', ..., 'id')))
```
Comment (pukkandan) [inline yt_dlp/extractor/iwara.py:138]:
```suggestion
        for x in traverse_obj(videos, ('results', ..., 'id')):
            yield self.url_result(f'https://iwara.tv/video/{x}')
```
`yield from ... for ...` is weird

File: yt_dlp/extractor/iwara.py [hunk 7]
```diff
@@ -1,239 +1,190 @@
...
+                functools.partial(self._entries, playlist_id, user_id),
+                self._PER_PAGE),
+            playlist_id, traverse_obj(user_info, ('user', 'name')))
 
 
-class IwaraUserIE(IwaraBaseIE):
-    _VALID_URL = fr'{IwaraBaseIE._BASE_REGEX}/users/(?P<id>[^/?#&]+)'
-    IE_NAME = 'iwara:user'
+class IwaraPlaylistIE(InfoExtractor):
+    # the ID is an UUID but I don't think it's necessary to write concrete regex
```
Comment (pukkandan) [inline yt_dlp/extractor/iwara.py:155]: I agree, but it may be useful to have a `UUID_RE` in utils

File: yt_dlp/extractor/iwara.py [hunk 8]
```diff
@@ -1,239 +1,190 @@
...
+        videos = first_page if page == 0 else self._download_json(
+            'https://api.iwara.tv/videos', playlist_id,
+            note=f'Downloading page {page}',
+            query={
+                'page': page,
+                'limit': self._PER_PAGE,
+            })
+        yield from (
+            self.url_result(f'https://iwara.tv/video/{x}')
+            for x in traverse_obj(videos, ('results', ..., 'id')))
```
Comment (pukkandan) [inline yt_dlp/extractor/iwara.py:178]:
```suggestion
        videos = self._download_json(
            'https://api.iwara.tv/videos', playlist_id, f'Downloading page {page}',
            query={'page': page, 'limit': self._PER_PAGE}) if page else first_page
        for x in traverse_obj(videos, ('results', ..., 'id')):
            yield self.url_result(f'https://iwara.tv/video/{x}')
```

**Review by pukkandan — APPROVED** [review]

File: yt_dlp/extractor/iwara.py (hunk 1 above)
Comment (Lesmiscore) [reply yt_dlp/extractor/iwara.py]: I think it _could_ be an empty string since that's what JS do but yeah

File: yt_dlp/extractor/iwara.py (hunk 3 above)
Comment (Lesmiscore) [reply yt_dlp/extractor/iwara.py]: Oh I meant `preference` there

**Comment by Lesmiscore** [comment]
@hasezoey example link?

**Comment by hasezoey** [comment]
> @hasezoey example link?

made a PR to fix the issue https://github.com/yt-dlp/yt-dlp/pull/6651

as for links which select the wrong quality, basically any that have `preview` as a format (meaning any newer video) and also `best` selecting `preview`

Example: `https://www.iwara.tv/video/lX5YcaGaCExlI2/iapopstars`

```sh
$ python3 ./yt_dlp/__main__.py -F -s https://www.iwara.tv/video/lX5YcaGaCExlI2/iapopstars
[iwara] Extracting URL: https://www.iwara.tv/video/lX5YcaGaCExlI2/iapopstars
[iwara] lX5YcaGaCExlI2: Downloading JSON metadata
[iwara] lX5YcaGaCExlI2: Downloading JSON metadata
[info] Available formats for lX5YcaGaCExlI2:
ID      EXT RESOLUTION │ PROTO │ VCODEC  ACODEC
────────────────────────────────────────────────
360     mp4 360p       │ https │ unknown unknown
540     mp4 540p       │ https │ unknown unknown
Source  mp4 unknown    │ https │ unknown unknown
preview mp4 unknown    │ https │ unknown unknown
[info] lX5YcaGaCExlI2: Downloading 1 format(s): preview

$ python3 ./yt_dlp/__main__.py -F -s -f best https://www.iwara.tv/video/lX5YcaGaCExlI2/iapopstars
WARNING: "-f best" selects the best pre-merged format which is often not the best option.
         To let yt-dlp download and merge the best available formats, simply do not pass any format selection.
         If you know what you are doing and want only the best pre-merged format, use "-f b" instead to suppress this warning
[iwara] Extracting URL: https://www.iwara.tv/video/lX5YcaGaCExlI2/iapopstars
[iwara] lX5YcaGaCExlI2: Downloading JSON metadata
[iwara] lX5YcaGaCExlI2: Downloading JSON metadata
[info] Available formats for lX5YcaGaCExlI2:
ID      EXT RESOLUTION │ PROTO │ VCODEC  ACODEC
────────────────────────────────────────────────
360     mp4 360p       │ https │ unknown unknown
540     mp4 540p       │ https │ unknown unknown
Source  mp4 unknown    │ https │ unknown unknown
preview mp4 unknown    │ https │ unknown unknown
[info] lX5YcaGaCExlI2: Downloading 1 format(s): preview
```

**Pull request merged**